src/mpl/src/mem/mpl_trmem.c

```diff
@@ -916,3 +916,16 @@ void *MPL_trstrdup(const char *str, int lineno, const char fname[])
     TR_THREAD_CS_EXIT;
     return retval;
 }
+
+void MPL_trcategorydump(FILE *fp)
+{
+    int i;
+
+    for (i = 0; i < MPL_MAX_MEMORY_CLASS; i++) {
+        fprintf(fp, "%d\t%ld\t%ld\t%ld\t%ld\n", i,
```

Comment:
something to convert class id to string would be really helpful here, IMO.

Reply:
Agreed. It'd just be another array and a conversion function. I can add it.